elixir-sdk/native/sdk_core/src/conversion.rs

```diff
@@ -0,0 +1,56 @@
+use rustler::{Encoder, Env, NifResult, Term};
+use std::collections::HashMap;
+use std::sync::Arc;
+use eppo_core::{AttributeValue, Str, events::AssignmentEvent, ufc::AssignmentValue};
+use rustler::types::atom;
+
+
+pub fn convert_attributes(subject_attributes: Term) -> NifResult<Arc<HashMap<Str, AttributeValue>>> {
+    // Obtain an iterator over the map's key-value pairs.
+    let map: HashMap<String, Term> = subject_attributes.decode()?;
```

Comment:
minor: would it make sense to use [`MapIterator`](https://docs.rs/rustler/latest/rustler/types/struct.MapIterator.html) here? Very minor, but `HashMap` allocates the structure and string keys ahead of time which then get discarded. It looks like we can use map iterator, so we don't need additional interim allocation

Another option is implementing `Decoder` for `Str` and `AttributeValue`, so we automatically get efficient implementation of `Decoder` for `HashMap<Str, AttributeValue>`